src/expr/mkmetakind

```diff
@@ -229,25 +229,37 @@ function constant {
     echo "$kf:$lineno: warning: constant $1 hasher \`$3' isn't fully-qualified (e.g., ::CVC4::RationalHashFunction)" >&2
   fi
 
-  if [ -n "$4" ]; then
+  # Avoid including the same header multiple times
+  if [ -n "$4" ] && [[ "${metakind_includes}" != *"#include \"$4\""* ]]; then
     metakind_includes="${metakind_includes}
 #include \"$4\""
   fi
   register_metakind CONSTANT "$1" 0
-  metakind_constantMaps="${metakind_constantMaps}
-}/* CVC4::kind::metakind namespace */
-}/* CVC4::kind namespace */
-
-namespace expr {
+  metakind_getConst_decls="${metakind_getConst_decls}
+template <>
+$2 const& NodeValue::getConst< $2 >() const;
+"
+  metakind_constantMaps_decls="${metakind_constantMaps_decls}
+template <>
```

Comment:
metakind_getConst_decls and metakind_constantMaps_decls are both in the header. I can add comments to make this clearer.